Rubberduck.Inspections/Concrete/SheetAccessedUsingStringInspection.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Inspections;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.Inspections.Resources;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Inspections.Concrete
+{
+    [RequiredLibrary("Excel")]
+    public class SheetAccessedUsingStringInspection : InspectionBase
+    {
+        public SheetAccessedUsingStringInspection(RubberduckParserState state) : base(state)
+        {
+        }
+
+        protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
+        {
+            var worksheetsDeclarations = BuiltInDeclarations.Where(decl =>
+                    decl.ProjectName.Equals("Excel") &&
```

Comment:
This doesn't appear to be sufficient to guarantee we're looking at *the* Excel type library. See how [`ImplicitActiveSheetReferenceInspection`](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Inspections/Concrete/ImplicitActiveSheetReferenceInspection.cs#L25) is looking for a not user-defined; adding a `!IsUserDefined` condition would be more robust IMO.

Reply:
FWIW, this may be an opportunity to make use of the TypeLib API as well... See:
https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibsAPI.cs#L407

Reply:
@retailcoder Adding `!IsUserDefined` here would be redundant since `BuildInDeclarations` is defined as `AllDeclarations.Where(declaration => !declaration.IsUserDefined)`.